org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/server/wizard/OrganizationProvider.java

```diff
@@ -0,0 +1,107 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.ui.internal.server.wizard;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jface.dialogs.IMessageProvider;
+import org.eclipse.jface.fieldassist.ContentProposal;
+import org.eclipse.jface.fieldassist.IContentProposal;
+import org.eclipse.jface.fieldassist.IContentProposalProvider;
+import org.eclipse.jface.operation.IRunnableWithProgress;
+import org.sonarlint.eclipse.core.SonarLintLogger;
+import org.sonarlint.eclipse.core.internal.server.IServer;
+import org.sonarsource.sonarlint.core.client.api.connected.RemoteOrganization;
+import org.sonarsource.sonarlint.core.client.api.exceptions.UnsupportedServerException;
+import org.sonarsource.sonarlint.core.client.api.util.TextSearchIndex;
+
+public class OrganizationProvider implements IContentProposalProvider {
+
+  private final ServerLocationWizardPage parentPage;
+  private String previousHost;
+  private String previousUsername;
+  private String previousPassword;
+  private TextSearchIndex<RemoteOrganization> orgs;
+
+  public OrganizationProvider(ServerLocationWizardPage parentPage) {
+    this.parentPage = parentPage;
+  }
+
+  @Override
+  public IContentProposal[] getProposals(String contents, int position) {
+    IServer transcientServer = parentPage.transcientServer();
+
+    if (!Objects.equals(previousHost, transcientServer.getHost())
+      || !Objects.equals(previousUsername, parentPage.getUsername())
+      || !Objects.equals(previousPassword, parentPage.getPassword())) {
+
+      previousHost = transcientServer.getHost();
+      previousUsername = parentPage.getUsername();
+      previousPassword = parentPage.getPassword();
+      try {
+        parentPage.getWizard().getContainer().run(false, false, new IRunnableWithProgress() {
+
+          @Override
+          public void run(IProgressMonitor monitor) throws InvocationTargetException, InterruptedException {
+            orgs = transcientServer.getOrganizationsIndex(parentPage.getUsername(), parentPage.getPassword());
+          }
+        });
+        parentPage.setMessage("", IMessageProvider.NONE);
+        return filtered(contents);
+      } catch (UnsupportedServerException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)